app/models/twitch_user.rb

```diff
@@ -3,8 +3,23 @@
 class TwitchUser < ApplicationRecord
   belongs_to :user
 
+  def self.from_auth(auth)
+    twitch_user = TwitchUser.find_or_initialize_by(twitch_id: auth.uid)
+    twitch_user.user = User.new if twitch_user.user.blank?
+    twitch_user.user.name = auth.info.nickname
+    twitch_user.assign_attributes(
+      access_token: auth.credentials.token,
+      name:         auth.info.nickname,
+      display_name: auth.info.name,
+      email:        auth.info.email,
+      avatar:       auth.info.image || TwitchUser.default_avatar
+    )
+    self.errors.add(twitch_user.user.errors) unless twitch_user.user.save
```

Comment:
Rubocop complains about redundant `self` here.  Also it doesn't appear that the new `TwitchUser` instance is ever getting saved, just the associated `User` object.